File: lang/typescript/lib/Select.ts
import {BaseIfc} from "./BaseIfc"

export abstract class Select extends BaseIfc {
    protected value : any
    toStepValue(isSelectOption : boolean = false) : string {{ return this.value.toStepValue(true); }}
    toSTEP() : string {{ return `${this.value.stepId} = ${this.constructor.name.toUpperCase()}(${this.value.getStepParameters()})` }}
}

File: lang/csharp/src/IFC/Select.cs
namespace IFC
{
    public abstract class Select : BaseIfc
    {
        protected BaseIfc choice;
        public BaseIfc Choice=>choice;
        public override string ToStepValue(bool isSelectOption = false){{ return choice.ToStepValue(true); }}
        public override string ToSTEP(){{ return $"#{choice.StepId} = {choice.GetType().Name.ToUpper()}({choice.GetStepParameters()});"; }}
    }
}